Gems/Camera/Code/Source/CameraSystemComponent.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "CameraSystemComponent.h"
+
+#include <AzCore/Math/MatrixUtils.h>
+#include <AzCore/Component/TransformBus.h>
+
+#include <AzFramework/Viewport/CameraState.h>
+
+#include <Atom/RPI.Public/View.h>
+#include <Atom/RPI.Public/ViewportContextBus.h>
+
+namespace Camera
+{
+    void CameraSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
```

Comment:
nit: With a cast like this as the type info is already on the right of the expression it's preferred to use `auto` here to remove redundancy